src/wasm2js.h

```diff
@@ -181,11 +187,23 @@ class Wasm2JSBuilder {
 
     // First up check our cached of mangled names to avoid doing extra work
     // below
-    auto& mangledScope = mangledNames[(int)scope];
-    auto it = mangledScope.find(name.c_str());
-    if (it != mangledScope.end()) {
+    auto& map = mangledNamesMap[(int)scope];
+    auto it = map.find(name.c_str());
+    if (it != map.end()) {
       return it->second;
     }
+    auto& set = mangledNamesSet[(int)scope];
+    // The Local scope is special: a Local name must not collide with a Top
+    // name, as they are in a single namespace in JS and can conflict:
+    //
+    // function foo(bar) {
+    //   var bar = 0;
+    // }
+    // function bar() { ..
+    std::unordered_set<IString>* additionalSet = nullptr;
```

Comment:
I think the logic would be easier to follow if this special casing of `NameScope::Local` were pushed down to the final check before warning rather than happening up here and requiring the `additionalSet` variable.

Reply:
Good point!